### PR TITLE
MULE-15540: Add utility functions to introspect the causes of an Error

### PR DESCRIPTION
**Pull request by alegmarra**

### CODE REVIEW

File: src/main/java/org/mule/extensions/java/internal/function/JavaModuleFunctions.java [hunk 1]
```diff
@@ -93,14 +97,68 @@ public Object invoke(@Alias("class") @Summary("Fully qualified name of the Class
...
+   * the specified class in the exception cause chain.
+   * If {@code acceptSubtypes} is {@code true}, subclasses of the specified class will also match.
+   *
+   * @param error           the error to inspect
+   * @param clazz           fully qualified name of the Class you want to check against
+   * @param includeSubtypes if true, subclasses of the specified class will also result in a match
+   * @return the index into the throwable chain, false if no match or null input
+   */
+  public boolean isCausedBy(Object error,
+                            @Alias("class") @Summary("Fully qualified name of the Class you want to check against") String clazz,
```
Comment (estebanwasinger) [inline src/main/java/org/mule/extensions/java/internal/function/JavaModuleFunctions.java]: shouldn't be `exception` \ `throwable` or `exceptionClass` \ `throwableClass` to be more declarative of what is expected to configure?

File: src/main/java/org/mule/extensions/java/internal/function/JavaModuleFunctions.java [hunk 2]
```diff
@@ -93,14 +96,43 @@ public Object invoke(@Alias("class") @Summary("Fully qualified name of the Class
...
    */
-  public boolean isInstanceOf(Object instance, @Alias("class") String clazz) throws ClassNotFoundModuleException {
+  public boolean isInstanceOf(Object instance,
+                              @Alias("class") @Summary("Fully qualified name of the Class you want to check against") String clazz)
+      throws ClassNotFoundModuleException {
     return isInstance(cache.loadClass(clazz), instance);
   }
 
+  /**
+   * Function that provides a way to obtain the root cause of a given {@link Error}.
```
Comment (afelisatti) [inline src/main/java/org/mule/extensions/java/internal/function/JavaModuleFunctions.java]: throwable

File: src/main/java/org/mule/extensions/java/internal/function/JavaModuleFunctions.java [hunk 3]
```diff
@@ -93,14 +96,43 @@ public Object invoke(@Alias("class") @Summary("Fully qualified name of the Class
...
+   * @param exception the {@link Throwable} whose root cause is wanted
+   * @return the root {@link Throwable cause} of the given {@code exception},
+   * or {@code null} if no cause is found.
+   */
+  public Throwable getRootCause(Throwable exception) {
+    return JavaErrorUtils.getRootCause(exception);
+  }
+
+  /**
+   * This Function returns {@code true} if the {@link Error} contains a {@link Throwable} that matches
```
Comment (afelisatti) [inline src/main/java/org/mule/extensions/java/internal/function/JavaModuleFunctions.java]: Throwable

File: src/main/java/org/mule/extensions/java/internal/util/JavaErrorUtils.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.java.internal.util;
+
+import java.util.function.BiFunction;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+/**
+ * Utility class for introspecting Mule Errors and their Java causes
```
Comment (afelisatti) [inline src/main/java/org/mule/extensions/java/internal/util/JavaErrorUtils.java]: same

**Pull request merged**